src/Filter/ChoiceFilter.php

```diff
@@ -89,18 +89,28 @@ private function filterWithMultipleValues(BaseProxyQueryInterface $query, string
         $completeField = sprintf('%s.%s', $alias, $field);
         $parameterName = $this->getNewParameterName($query);
         if (EqualOperatorType::TYPE_NOT_EQUAL === $data['type']) {
-            $andConditions = [$query->getQueryBuilder()->expr()->isNotNull($completeField)];
+            $or = $query->getQueryBuilder()->expr()->orX();
```

Comment:
The following lines are hard to read/understand. Can you try to extract some smaller methods?

Reply:
The code is simpler now and doesn't need others private methods.